tools/image-builder/build_bond.zsh

```diff
@@ -0,0 +1,82 @@
+#!/bin/zsh
+
+set -eux
+setopt shwordsplit
+
+HOME=$1
+FLAVOR=$2
+BOOST=${3:-}
+
+BUILD_PATH=/root/build/$FLAVOR
+if [ $BOOST != "" ]; then BUILD_PATH=$BUILD_PATH/$BOOST; fi
+
+mkdir -p $HOME $BUILD_PATH
+ln -s /root/.stack $HOME/.stack
+
+cd $BUILD_PATH
+
+case "$FLAVOR" in
+    cpp-*)
+        case "$FLAVOR" in
+            "cpp-core") CPP_CMAKE_ARGS="-DBOND_SKIP_GBC_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE";;
+            "cpp-comm") CPP_CMAKE_ARGS="-DBOND_ENABLE_COMM=TRUE -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE";;
+            "cpp-grpc") CPP_CMAKE_ARGS="-DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DgRPC_ZLIB_PROVIDER=package";;
+            *) echo "Unknown FLAVOR=$FLAVOR"; exit 1;;
+        esac
+
+        export BOOST_ROOT=/opt/boosts/boost_`echo $BOOST | tr . _`
+        export CXX="ccache clang++ -Qunused-arguments --system-header-prefix=boost/"
+        export CC="ccache clang -Qunused-arguments --system-header-prefix=boost/"
+
+        ln -s /root/.ccache $HOME/.ccache
+
+        cmake $CPP_CMAKE_ARGS /root/bond
+
+        make --jobs 2 check
+        ;;
+
+    "cs")
+        # TODO: Remove build dependency on C++
+        export BOOST_ROOT=/opt/boosts/boost_1_63_0
+        export CXX="ccache clang++ -Qunused-arguments --system-header-prefix=boost/"
+        export CC="ccache clang -Qunused-arguments --system-header-prefix=boost/"
+        ln -s /root/.ccache $HOME/.ccache
+
+        nuget restore /root/bond/cs/cs.sln
+
+        cmake -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE /root/bond
+
+        make gbc
+        make DESTDIR=/root install
```

Comment:
The boosts have mutually exclusive install locations, so they make sense in `/opt`, and the NUnit runner doesn't really belong... anywhere, but it seems like letting `make install` put `gbc` in `/usr/local/bin` would be appropriate. That gets `gbc` into the default `$PATH`, which will definitely simplify Java and probably simplify C#.

Reply:
Makes sense. Updated.